cyw43/src/bus.rs

```diff
@@ -196,28 +244,45 @@ where
     }
 
     async fn backplane_readn(&mut self, addr: u32, len: u32) -> u32 {
+        defmt::debug!("backplane_readn addr = {:08x} len = {}", addr, len);
+
         self.backplane_set_window(addr).await;
 
         let mut bus_addr = addr & BACKPLANE_ADDRESS_MASK;
         if len == 4 {
-            bus_addr |= BACKPLANE_ADDRESS_32BIT_FLAG
+            bus_addr |= BACKPLANE_ADDRESS_32BIT_FLAG;
         }
-        self.readn(FUNC_BACKPLANE, bus_addr, len).await
+        
+        let val = self.readn(FUNC_BACKPLANE, bus_addr, len).await;
+
+        defmt::debug!("backplane_readn addr = {:08x} len = {} val = {:08x}", addr, len, val);
+
+        self.backplane_set_window(0x18000000).await; // CHIPCOMMON_BASE_ADDRESS
+
+        return val
     }
 
     async fn backplane_writen(&mut self, addr: u32, val: u32, len: u32) {
+        defmt::debug!("backplane_writen addr = {:08x} len = {} val = {:08x}", addr, len, val);
+
         self.backplane_set_window(addr).await;
 
         let mut bus_addr = addr & BACKPLANE_ADDRESS_MASK;
         if len == 4 {
-            bus_addr |= BACKPLANE_ADDRESS_32BIT_FLAG
+            bus_addr |= BACKPLANE_ADDRESS_32BIT_FLAG;
         }
-        self.writen(FUNC_BACKPLANE, bus_addr, val, len).await
+        self.writen(FUNC_BACKPLANE, bus_addr, val, len).await;
+
+        self.backplane_set_window(0x18000000).await; // CHIPCOMMON_BASE_ADDRESS
     }
 
     async fn backplane_set_window(&mut self, addr: u32) {
         let new_window = addr & !BACKPLANE_ADDRESS_MASK;
 
+        if new_window == self.backplane_window {
```

Comment:
i think this is a big win? if you think for some reason it's a bad idea i can get rid of it

Reply:
if the window is unchanged, the 3 if's below will already do nothing.

Reply:
will get rid of it, thanks